### PR TITLE
fed prox example

### PR DESCRIPTION
**Pull request by mattiasakesson**

### CODE REVIEW

File: examples/mnist-pytorch-fedprox/client/entrypoint [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+            batch_x = x_train[b * batch_size:(b + 1) * batch_size]
+            batch_y = y_train[b * batch_size:(b + 1) * batch_size]
+            # Train on batch
+            optimizer.zero_grad()
+            outputs = model(batch_x)
+
+            proximal_term = 0.0
+            for w, w_t in zip(model.parameters(), global_model.parameters()):
+                proximal_term += (w - w_t).norm(2)
+            #print("proximal_term: ", proximal_term)
```
Comment (ahellander) [inline examples/mnist-pytorch-fedprox/client/entrypoint:168]: Clean up comments

**Review by ahellander — CHANGES_REQUESTED** [review]
Looks clean and neat. Would it be possible to have a notebook or script that runs vanilla fedavg and fedprox?  I guess we would gave to have some significant degree of heterogeneity to have prox show an advnatage?

**Comment by mattiasakesson** [comment]
Updated.
I didnt add a notbok instead a wrote some lines in the readme about how to use fedprox and how to turn the example to vanilla fedavg by setting mu=0. I dont think a notebok will make it any clearer, you still have to manually modify the client_settings.yaml file to change the mu parameter.

**Comment by ahellander** [comment]
Will it / how will it work with client_settings.yaml in the base directory and not in  the 'client' folder? 

Also, we need to rework to align with the new model incoming from the "python_env" PR.

**Comment by ahellander** [comment]
Given the release with the python_env support, this one needs to be updated to the new client structure.

**Comment by ahellander** [comment]
@mattiasakesson lets get the fedprox example finished, I think it is good to show how it can work.

**Comment by Wrede** [comment]
Will close due to inactivity, can be re-open at later stage.

**Pull request closed without merging**